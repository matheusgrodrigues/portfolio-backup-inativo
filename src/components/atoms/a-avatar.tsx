import { styled } from "styled-components";

import * as RadixAvatar from "@radix-ui/react-avatar";

interface IAvatarRoot extends RadixAvatar.AvatarProps {
  "data-testid"?: string;
}

const AvatarRoot = styled(RadixAvatar.Root).attrs<IAvatarRoot>(
  (props) => ({
    as: "span",
    "data-testid": props["data-testid"] || "a-avatar",
  })
)`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  vertical-align: middle;
  overflow: hidden;
  user-select: none;
  width: 45px;
  height: 45px;
  border-radius: 100%;
  background-color: var(--black-a3);
`;

interface IAvatarImage extends RadixAvatar.AvatarImageProps {
  "data-testid"?: string;
}

const AvatarImage = styled(RadixAvatar.Image).attrs<IAvatarImage>(
  (props) => ({
    as: "img",
    "data-testid": props["data-testid"] || "a-avatar-image",
  })
)`
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: inherit;
`;

interface IAvatarFallBack extends RadixAvatar.AvatarFallbackProps {
  "data-testid"?: string;
}

const AvatarFallback = styled(
  RadixAvatar.Fallback
).attrs<IAvatarFallBack>((props) => ({
  as: "span",
  "data-testid": props["data-testid"] || "a-avatar-fallback",
}))`
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: white;
  color: var(--violet-11);
  font-size: 15px;
  line-height: 1;
  font-weight: 500;
`;

interface IAvatar extends RadixAvatar.AvatarProps {
  src: string;
  alt: string;
}

export function Avatar({ src, alt, ...props }: IAvatar) {
  return (
    <AvatarRoot {...props}>
      <AvatarImage src={src} alt={alt} />
    </AvatarRoot>
  );
}
