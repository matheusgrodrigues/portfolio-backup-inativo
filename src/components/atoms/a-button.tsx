import styled from "styled-components";

interface IAButton {
  $variant: "lg";
  $background: "color_white" | "color_gradient_primary600";
  $fontSize: "fs_textMd";
  $fontWeight: "fw_semibold";
  $color: "color_white" | "color_gray500";
}

export const AButton = styled.button<IAButton>`
  width: 100%;
  border-radius: 8px;
  border: none;
  cursor: pointer;

  ${(props) => {
    switch (props.$variant) {
      case "lg":
        return `
            max-width: 172px;
            height: 44px;
            `;
        break;
    }
  }}

  background: ${(props) => props.theme.ref.colors[props.$background]};
  color: ${(props) => props.theme.ref.colors[props.$color]};
  font-weight: ${(props) => props.theme.ref.font[props.$fontWeight]};
  font-size: ${(props) => props.theme.ref.font[props.$fontSize]};
`;
