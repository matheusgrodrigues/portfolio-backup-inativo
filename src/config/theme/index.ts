import { colors, colorsDark, font, spacing, bp } from './ref'

export const bpHelper = (bp: string, content: string) => {
    return `
  @media screen and (min-width: ${bp}) {
    ${content}
  }
  `
}

export const themeLight = {
    ref: { colors, font, spacing, bp },
}

export const darkTheme = {
    ...themeLight,
    ref: { ...themeLight.ref, colors: { ...colorsDark } },
}
