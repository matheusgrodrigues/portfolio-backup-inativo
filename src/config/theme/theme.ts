const fontFamily = {
    fontFamily_inter: 'Inter',
};

export type FontSizeName = 'fontSize_textSm' | 'fontSize_textMd' | 'fontSize_textXl' | 'fontSize_displayLg';

const fontSize = {
    fontSize_textSm: '14px',
    fontSize_textMd: '16px',
    fontSize_textXl: '20px',
    fontSize_displayLg: '48px',
};

export type FontWeightName = 'fontWeight_semibold' | 'fontWeight_regular' | 'fontWeight_medium';

const fontWeight = {
    fontWeight_regular: '400',
    fontWeight_medium: '500',
    fontWeight_semibold: '600',
};

export type ColorName =
    | 'color_gradient_primary600'
    | 'color_primary500'
    | 'color_primary600'
    | 'color_gray500'
    | 'color_gray900'
    | 'color_gray50'
    | 'color_white'
    | 'color_black'
    | 'color_transparent';

const colors = {
    color_gradient_primary600: `linear-gradient(90deg, #7f56d9 0%, #9e77ed 100%);`,
    color_primary500: '#9e77ed',
    color_primary600: '#7f56d9',
    color_gray500: '#667085',
    color_gray900: '#101828',
    color_gray50: '#f9fafb',
    color_white: '#ffffff',
    color_black: '#000000',
    color_transparent: 'transparent',
};

export type SpacingName =
    | 'spacing_4'
    | 'spacing_8'
    | 'spacing_12'
    | 'spacing_16'
    | 'spacing_20'
    | 'spacing_24'
    | 'spacing_32'
    | 'spacing_40'
    | 'spacing_48'
    | 'spacing_64'
    | 'spacing_80'
    | 'spacing_96'
    | 'spacing_128'
    | 'spacing_160'
    | 'spacing_192'
    | 'spacing_224'
    | 'spacing_256';

const spacing = {
    spacing_4: '4px',
    spacing_8: '8px',
    spacing_12: '12px',
    spacing_16: '16px',
    spacing_20: '20px',
    spacing_24: '24px',
    spacing_32: '32px',
    spacing_40: '40px',
    spacing_48: '48px',
    spacing_64: '64px',
    spacing_80: '80px',
    spacing_96: '96px',
    spacing_128: '128px',
    spacing_160: '160px',
    spacing_192: '192px',
    spacing_224: '224px',
    spacing_256: '256px',
};

type BreakpointName = 'breakpoint_sm' | 'breakpoint_md' | 'breakpoint_lg' | 'breakpoint_xl';

const breakpoint = {
    breakpoint_sm: '640px',
    breakpoint_md: '768px',
    breakpoint_lg: '1024px',
    breakpoint_xl: '1280px',
};

export const bpHelper = (bp: BreakpointName, content: string) =>
    `@media screen and (min-width: ${breakpoint[bp]}) {
    ${content}
  }`;

export type ThemeName = 'light' | 'dark';

type Theme = {
    name: ThemeName;
    ref: {
        breakpoint: typeof breakpoint;
        fontFamily: typeof fontFamily;
        fontWeight: typeof fontWeight;
        fontSize: typeof fontSize;
        spacing: typeof spacing;
        colors: typeof colors;
    };
};

declare module 'styled-components' {
    export interface DefaultTheme extends Theme {}
}

export const themeLight = {
    name: 'light',
    ref: { breakpoint, fontFamily, fontWeight, fontSize, spacing, colors },
};

export const darkTheme = {
    name: 'dark',
    ref: { breakpoint, fontFamily, fontWeight, fontSize, spacing, colors },
};
