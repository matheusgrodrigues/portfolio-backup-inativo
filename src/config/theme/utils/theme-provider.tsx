'use client'

import { ThemeProvider as StyledThemeProvider } from 'styled-components'

export const ThemeProvider = StyledThemeProvider
