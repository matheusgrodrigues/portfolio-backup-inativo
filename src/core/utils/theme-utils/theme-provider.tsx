'use client';

import { ThemeProvider as StyledThemeProvider } from 'styled-components';

const ThemeProvider = StyledThemeProvider;

export default ThemeProvider;
