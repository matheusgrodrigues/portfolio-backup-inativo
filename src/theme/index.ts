import { colors, colorsDark, font } from "./ref";

export const themeLight = {
  ref: { colors, font },
};

export const darkTheme = {
  ...themeLight,
  ref: { ...themeLight.ref, colors: { ...colorsDark } },
};
