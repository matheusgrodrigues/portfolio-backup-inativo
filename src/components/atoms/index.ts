import { AAvatar } from "./a-avatar/a-avatar";

export { AAvatar };
