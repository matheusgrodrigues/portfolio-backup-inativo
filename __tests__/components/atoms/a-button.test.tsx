import renderer from "react-test-renderer";

// Atoms
import { AButton } from "@/src/components/atoms/a-button";

// Theme
import { ThemeProvider } from "@/src/theme/helpers/theme-provider";
import { themeLight } from "@/src/theme";

describe("Deve renderizar o AButton corretamente", () => {
  it("Deve preservar a estrutura visual do componente", () => {
    const three = renderer
      .create(
        <ThemeProvider theme={themeLight}>
          <AButton
            $variant="lg"
            $background="color_gradient_primary600"
            $fontSize="fs_textMd"
            $fontWeight="fw_semibold"
            $color="color_gray500"
          >
            AButton
          </AButton>
        </ThemeProvider>
      )
      .toJSON();

    expect(three).toMatchSnapshot();
  });
});
