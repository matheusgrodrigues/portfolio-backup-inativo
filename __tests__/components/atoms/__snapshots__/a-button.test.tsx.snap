// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o AButton corretamente Deve preservar a estrutura visual do componente 1`] = `
<button
  className="sc-beyTiQ eBOnYr"
>
  AButton
</button>
`;
